api/sparse/views.py

```diff
@@ -0,0 +1,88 @@
+from rest_framework.exceptions import MethodNotAllowed
+from api.sparse.serializers import SparseNodeSerializer, SparseRegistrationSerializer
+from api.nodes.views import (
+    NodeDetail,
+    NodeChildrenList,
+    NodeList,
+    LinkedNodesList,
+    NodeLinkedRegistrationsList,
+)
+
+from api.registrations.views import RegistrationDetail, RegistrationChildrenList, RegistrationList
+from api.users.views import UserNodes, UserRegistrations
+
+from api.base.pagination import CursorPagination
+
+
+class BaseSparseMixin(object):
+    view_category = 'sparse'
+
+    pagination_class = CursorPagination
+    serializer_class = None
+
+    # overrides NodeList because these endpoints don't allow writing
+    def perform_create(self, *args):
+        raise MethodNotAllowed(method=self.request.method)
+
+    # overrides NodeList because these endpoints don't allow writing
+    def perform_update(self, *args):
+        raise MethodNotAllowed(method=self.request.method)
+
+    # overrides NodeDetail because these endpoints don't allow writing
+    def perform_destroy(self, *args):
+        raise MethodNotAllowed(method=self.request.method)
+
+    # overrides NodeList because these endpoints don't allow writing
+    def allow_bulk_destroy_resources(self, *args):
+        raise MethodNotAllowed(method=self.request.method)
+
+    def get_serializer_class(self):
+        return self.serializer_class
+
+
+class SparseNodeMixin(BaseSparseMixin):
+    serializer_class = SparseNodeSerializer
+
+
+class SparseRegistrationMixin(BaseSparseMixin):
+    serializer_class = SparseRegistrationSerializer
+
+
```

Comment:
A `TODO` for the following classes could be to add docstrings. They are currently inherited from the non-sparse classes and link to the documentation for the standard api endpoint (which makes complete sense because we don't have documentation for these new sparse endpoints).

Reply:
In the browseable API? Yeah, that makes sense.

Reply:
It was more just a minor thing I noticed in my text editor (VS Code) but the implications for the Browseable API is a better reason.